File: fib2.r
fib <- function(n){
	a0 = 0;
	a1 = 1;
	an = 0;
	if(n==0){
		return(a0);
	}else if(n==1){
		return(a1);
	}else{
		for(i in 2:n){
			an = a0 + a1;
			a0 = a1;
			a1 = an;
		}
		return(an);
	}
}

File: fib2.rb
#!/usr/bin/ruby

def fib(0)
	return(0)
end
def fib(1)
	return(1)
end
def fib(n)
	return(fib(n-1) + fib(n-2))
end

vin = ARGV[0].to_i
out = fib(vin)
print(out)
print("\n")
